docs/ovs-offload.md

```diff
@@ -0,0 +1,201 @@
+are required. Please make sure you have root privileges to run the commands
+below.
+
+Check the Number of VF Supported on the NIC
+
+```
+cat /sys/class/net/enp3s0f0/device/sriov_totalvfs
+8
+```
+
+Create the VFs
+
+```
+echo '4' > /sys/class/net/enp3s0f0/device/sriov_numvfs
+```
+
+Verfiy that the VFs are created
+
+```
+ip link show enp3s0f0
+8: enp3s0f0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT qlen 1000
+   link/ether a0:36:9f:8f:3f:b8 brd ff:ff:ff:ff:ff:ff
+   vf 0 MAC 00:00:00:00:00:00, spoof checking on, link-state auto
+   vf 1 MAC 00:00:00:00:00:00, spoof checking on, link-state auto
+   vf 2 MAC 00:00:00:00:00:00, spoof checking on, link-state auto
+   vf 3 MAC 00:00:00:00:00:00, spoof checking on, link-state auto
+```
+
+Set up the PF to be up
+
+```
+ip link set enp3s0f0 up
+```
+
+Unbind the VFs from the driver
+
+```
+echo 0000:03:00.2 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.3 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.4 > /sys/bus/pci/drivers/mlx5_core/unbind
+echo 0000:03:00.5 > /sys/bus/pci/drivers/mlx5_core/unbind
+```
+
+Configure SR-IOV VFs to switchdev mode
+
+```
+devlink dev eswitch set pci/0000:03:00.0 mode switchdev
+ethtool -K enp3s0f0 hw-tc-offload on
+```
+
+Bind the VFs to the driver
+
+```
+echo 0000:03:00.2 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.3 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.4 > /sys/bus/pci/drivers/mlx5_core/bind
+echo 0000:03:00.5 > /sys/bus/pci/drivers/mlx5_core/bind
+```
+
+## SR-IOV network device plugin configuration
+
+Create a ConfigMap that defines SR-IOV resrouce pool configuration
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: sriovdp-config
+  namespace: kube-system
+data:
+  config.json: |
+    {
+      "resourceList": [{
+          "resourcePrefix": "mellanox.com",
+          "resourceName": "cx5_sriov_switchdev",
+          "isRdma": true,
+          "selectors": {
+                  "vendors": ["15b3"],
+                  "devices": ["1018"],
+                  "drivers": ["mlx5_core"]
+              }
+      }
+      ]
+    }
+```
+
+Deploy SR-IOV network device plugin as DaemonSet see https://github.com/intel/sriov-network-device-plugin
+
+Deploy multus CNI as DaemonSet see https://github.com/intel/multus-cni
+
+Create NetworkAttachementDefinition CRD with Antrea CNI config
+
+```yaml
+Kubernetes Network CRD Spec:
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: default
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: mellanox.com/cx5_sriov_switchdev
+spec:
+  config: '{
+    "cniVersion": "0.3.1",
+    "name": "antrea",
+    "plugins": [ { "type": "antrea", "ipam": { "type": "host-local" } }, { "type": "portmap", "capabilities": {"portMappings": true} }]
+}'
+```
+## Deploy Antrea Image with hw-offload enabled
+Modify the build/yamls/antrea.yml with offload flag
+
+```yaml
+  - command:
+    - start_ovs
+    - offload
```

Comment:
Have we considered an environment variable as something more user friendly here?
@tnqn @jianjuns

Reply:
We do not use environment much. I would not introduce multiple ways to pass config options either.

Reply:
I still prefer command argument. @antoninbas ?